### PR TITLE
Add drag and drop for groups -> groups

### PR DESCRIPTION
**Pull request by tobiasdiez**
Enable drag and drop in the group panel.
I also had to remove the "worker" aspect of the group worker, ie. now the entry list is updated immediately (this might lead to performance problems in bigger db's; but without this change JabRef would just hang after drop and down).

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)
- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I will add my changes from the other PR to this one

**Comment by matthiasgeiger** [comment]
@Siedlerchr Really? Isn't the scope of this PR different? 
You are working on "assign entries to groups by drag-and-drop" whereas this PR deals with "restructure the groups tree with drag-and-drop".

However, it yours and @tobiasdiez's decision whether this makes sense from the technical aspect.

**Comment by Siedlerchr** [comment]
In fact it uses the same methods for listening/reaction of an DND Event. My changes would be simply an else case.

**Comment by matthiasgeiger** [comment]
Allright, then forget my objections :wink:

**Comment by Siedlerchr** [comment]
@tobiasdiez  Drag and drop of entries works now as expected 🥇

File: src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java [hunk 1]
```diff
@@ -67,7 +67,8 @@ public StringProperty filterTextProperty() {
      * We need to notify the {@link StateManager} about this change so that the main table gets updated.
      */
     private void onSelectedGroupChanged(GroupNodeViewModel newValue) {
-        if (currentDatabase != stateManager.activeDatabaseProperty().get()) {
+        //TODO: Can one of this be empty?
+        if (currentDatabase.get() != stateManager.activeDatabaseProperty().get().get()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java]: @tobiasdiez  I am not sure if we should compare the values or if it is enough to check the object refs

Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java]: I think `equals` is the cleanest solution

**Comment by tobiasdiez** [comment]
Thanks @Siedlerchr for making the groups great again!

File: src/main/java/org/jabref/gui/groups/GroupTreeController.java [hunk 2]
```diff
@@ -136,9 +136,54 @@ public void initialize() {
...
+                    if (source.isPresent()) {
+                        source.get().moveTo(row.getItem());
+                        success = true;
+                    }
+                }
+                if (dragboard.hasContent(DragAndDropDataFormats.ENTRIES)) {
+                    TransferableEntrySelection entrySelection = (TransferableEntrySelection) dragboard
+                            .getContent(DragAndDropDataFormats.ENTRIES);
+
+                    row.getItem().addEntriesToGroup(entrySelection.getSelection());
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupTreeController.java]: Forgot to set succes true in the entries else part

Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupTreeController.java]: Pushed to master

**Comment by matthiasgeiger** [comment]
Short question here: Drag-and-drop generally works in current `master` - but the main table is not updated at once. I.e., after dropping an entry to the currently selected group it is still grayed out as an "non-member". After reselecting the group it is displayed correctly in the "floating" list of group members.

Is this a known problem? Or should I open an issue?

**Comment by tobiasdiez** [comment]
I added it to the list of open group-related problems: https://github.com/JabRef/jabref/issues/2599

**Pull request merged**